doc/modules/clustering.rst

```diff
@@ -1582,3 +1582,32 @@ Drawbacks
  *  Caliński, T., & Harabasz, J. (1974). "A dendrite method for cluster
     analysis". Communications in Statistics-theory and Methods 3: 1-27.
     `doi:10.1080/03610926.2011.560741 <http://dx.doi.org/10.1080/03610926.2011.560741>`_.
+
+.. _contingency_matrix:
+
+Contingency Matrix
+----------------------
+
+Contingency matrix reports the intersection cardinality for every
+true/predicted cluster pair. The contingency matrix provides sufficient
+statistics for all clustering metrics where the samples are
+independent and identically distributed and one doesn't need to account
+for some instances not being clustered.
+
+Here is an example::
+
+   >>> from sklearn.metrics.cluster import contingency_matrix
+   >>> x = ["male", "male", "male", "female", "female", "female"]
```

Comment:
In the context of clustering evaluation, I find it strange to use meaningful labels. In this case, the order of rows and columns *is* significant, so it's not really what we're going for.

Reply:
Wouldn't the ordering of rows and columns be always significant in a contingency matrix? (Not just in scikit learn but even in general).